testing/src/main/java/io/grpc/internal/testing/AbstractTransportTest.java

```diff
@@ -0,0 +1,907 @@
+  }
+
+  @Test
+  public void earlyServerClose_noServerHeaders() throws Exception {
+    server.start(serverListener);
+    client.start(mockClientTransportListener);
+    MockServerTransportListener serverTransportListener
+        = serverListener.takeListenerOrFail(1000, TimeUnit.MILLISECONDS);
+    serverTransport = serverTransportListener.transport;
+
+    ClientStream clientStream = client.newStream(methodDescriptor, new Metadata());
+    clientStream.start(mockClientStreamListener);
+    StreamCreation serverStreamCreation
+        = serverTransportListener.takeStreamOrFail(1000, TimeUnit.MILLISECONDS);
+    ServerStream serverStream = serverStreamCreation.stream;
+    ServerStreamListener mockServerStreamListener = serverStreamCreation.listener;
+
+    Status status = Status.OK.withDescription("Hellogoodbye");
+    Metadata trailers = new Metadata();
+    trailers.put(asciiKey, "trailers");
+    trailers.put(asciiKey, "dupvalue");
+    trailers.put(asciiKey, "dupvalue");
+    trailers.put(binaryKey, "äbinarytrailers");
+    serverStream.close(status, trailers);
+    verify(mockServerStreamListener, timeout(1000)).closed(statusCaptor.capture());
+    assertCodeEquals(Status.OK, statusCaptor.getValue());
+    verify(mockClientStreamListener, timeout(1000))
+        .closed(statusCaptor.capture(), metadataCaptor.capture());
+    assertEquals(status.getCode(), statusCaptor.getValue().getCode());
+    assertEquals(Lists.newArrayList(trailers.getAll(asciiKey)),
+        Lists.newArrayList(metadataCaptor.getValue().getAll(asciiKey)));
+    assertEquals(Lists.newArrayList(trailers.getAll(binaryKey)),
+        Lists.newArrayList(metadataCaptor.getValue().getAll(binaryKey)));
+  }
+
+  @Test
+  public void earlyServerClose_serverFailure() throws Exception {
+    server.start(serverListener);
+    client.start(mockClientTransportListener);
+    MockServerTransportListener serverTransportListener
+        = serverListener.takeListenerOrFail(1000, TimeUnit.MILLISECONDS);
+    serverTransport = serverTransportListener.transport;
+
+    ClientStream clientStream = client.newStream(methodDescriptor, new Metadata());
+    clientStream.start(mockClientStreamListener);
+    StreamCreation serverStreamCreation
+        = serverTransportListener.takeStreamOrFail(1000, TimeUnit.MILLISECONDS);
+    ServerStream serverStream = serverStreamCreation.stream;
+    ServerStreamListener mockServerStreamListener = serverStreamCreation.listener;
+
+    Status status = Status.INVALID_ARGUMENT.withDescription("I'm not listening");
+    serverStream.close(status, new Metadata());
+    verify(mockServerStreamListener, timeout(1000)).closed(statusCaptor.capture());
+    assertCodeEquals(Status.OK, statusCaptor.getValue());
+    verify(mockClientStreamListener, timeout(1000))
+        .closed(statusCaptor.capture(), any(Metadata.class));
+    assertEquals(status.getCode(), statusCaptor.getValue().getCode());
```

Comment:
Yeah. Generally I don't want to check description every place I could, just places it seems more likely for something to go wrong. But there isn't another test verify the status in this case so it does make sense to verify the description.